### PR TITLE
Add ability to "favorite" operations

### PR DESCRIPTION
**Pull request by TylerNoblett**
This PR gives users the ability to 'favorite' an operation. 

The only feature this enables right now is that the favorite and non-fav operations are separated into two different groups on the main screen.

A user can favorite the operation on the main screen, or when viewing a specific operation.

### No Favorites 
You'd see almost the exact same view if all operations were favorited - this is basically the same as before
Screen Shot 2022-09-15 at 11 56 34 AM

### With Favorites
Screen Shot 2022-09-15 at 11 56 24 AM

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: backend/database/seeding/helpers.go [hunk 1]
```diff
@@ -228,12 +228,18 @@ func newFindingGen(first int64) func(opID int64, uuid string, category *int64, t
 	}
 }
 
-func newUserOpPermission(user models.User, op models.Operation, role policy.OperationRole) models.UserOperationPermission {
+func newUserOpPermission(user models.User, op models.Operation, role policy.OperationRole, favorite ...bool) models.UserOperationPermission {
+	isFavorite := false
+	if len(favorite) > 0 {
+		isFavorite = true
+	}
```
Comment (JoelAtDeluxe) [inline backend/database/seeding/helpers.go:235]:
```suggestion
	isFavorite := (len(favorite) > 0)
```

Comment (JoelAtDeluxe) [reply backend/database/seeding/helpers.go:235]: or you could put it in the structure creation

File: frontend/src/pages/operation_list/operation_card/stylesheet.styl [hunk 2]
```diff
@@ -54,6 +54,43 @@
   &:hover
     background-color: $darker-background
 
+.favorite
```
Comment (JoelAtDeluxe) [inline frontend/src/pages/operation_list/operation_card/stylesheet.styl:57]: My CSS skills aren't great, but I think we can reduce the duplication here.

1. Keep `.not-favorite` (consider a rename -- you'll see why in a bit)
2. Rename `favorite` and remove everything but `background-image: url('./star-solid.svg')`
3. in the operation_card/index.tsx, changed the classname as such:
`className={cx('not-favorite', isFavorite && 'favorite')}`

For what it's worth, I renamed these to `favorite-button` (with the unfilled star), and `filled`. Here's what my final result looks like:

```css
.favorite-button
  position: absolute
  top: 70px
  right: 15px
  width: 25px
  min-width: 25px
  height: @width
  background-color: $background
  background-position: center center
  background-size: 12px
  background-repeat: no-repeat
  background-image: url('./star-outline.svg')
  box-shadow: inset 0 0 0 1px $darker-background
  border-radius: 3px
  padding: 0px

  &.filled
    background-image: url('./star-solid.svg')

  &:hover
    background-color: $darker-background
```

```tsx
<Button
        className={cx('favorite-button', isFavorite && 'filled')}
        onClick={() => setIsFavorite(!isFavorite)}>
      </Button>
```

File: frontend/src/pages/operation_show/layout/sidebar/stylesheet.styl [hunk 3]
```diff
@@ -65,6 +65,55 @@
     opacity: 1
     background-color: $background
 
+.favorite
```
Comment (JoelAtDeluxe) [inline frontend/src/pages/operation_show/layout/sidebar/stylesheet.styl:68]: same note here with the css change (though I didn't test this one out)

File: backend/database/seeding/helpers.go (hunk 1 above)
Comment (TylerNoblett) [reply backend/database/seeding/helpers.go:235]: ah yeah that's much nicer - I think I'll put it in the struct

File: backend/database/seeding/test_helpers.go [hunk 4]
```diff
@@ -464,6 +464,37 @@ func GetUsersWithRoleForOperationByOperationID(t *testing.T, db *database.Connec
...
+	err := db.Get(&operationID, sq.Select("id").
+		From("operations").
+		Where(sq.Eq{"slug": slug}))
+	require.NoError(t, err)
+
+	var isFavorite bool
+	err = db.Get(&isFavorite, sq.Select("is_favorite").
+		From("user_operation_permissions").
+		Where(sq.Eq{"operation_id": operationID, "user_id": id}))
+	require.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (TylerNoblett) [reply backend/database/seeding/test_helpers.go:493]: ah yeah that's good

File: frontend/src/pages/operation_list/operation_card/stylesheet.styl (hunk 2 above)
Comment (TylerNoblett) [reply frontend/src/pages/operation_list/operation_card/stylesheet.styl]: oh duh - I'm so using to working with JS-within-css (like styled components) that I forgot how to de-dup css the actual way 😅

File: frontend/src/pages/operation_show/layout/sidebar/stylesheet.styl (hunk 3 above)
Comment (TylerNoblett) [reply frontend/src/pages/operation_show/layout/sidebar/stylesheet.styl]: changed this as well

File: frontend/src/pages/operation_list/index.tsx [hunk 5]
```diff
@@ -26,9 +27,63 @@ export default () => {
...
+      if (data) {
+        const [ops, welcomeFlag] = data
+        setOps(ops)
+        setWelcomeFlag(welcomeFlag)
+      }
+    })
+  }, [wiredData])
+
+  type Header = "Other" | "Favorites" | null
+  const showOperationList = (ops: Operation[], header: Header) => (
```
Comment (JoelAtDeluxe) [inline frontend/src/pages/operation_list/index.tsx:44]: This works, but I'd consider moving this into a stand alone component.

**Review by JoelAtDeluxe — APPROVED** [review]
Looks good. I like the expose function you've added.It helps with some of the complicated rendering logic that used to be forced into the render method.

File: frontend/src/pages/operation_list/index.tsx (hunk 5 above)
Comment (TylerNoblett) [reply frontend/src/pages/operation_list/index.tsx:44]: for sure!

**Comment by TylerNoblett** [comment]
> Looks good. I like the expose function you've added.It helps with some of the complicated rendering logic that used to be forced into the render method.

thanks! Yeah that was my intent

File: frontend/src/pages/operation_list/index.tsx (hunk 5 above)
Comment (TylerNoblett) [reply frontend/src/pages/operation_list/index.tsx]: changed this into a component

Comment (TylerNoblett) [reply frontend/src/pages/operation_list/index.tsx]: actually fixing a bug rn that I created when I made this change

Comment (TylerNoblett) [reply frontend/src/pages/operation_list/index.tsx]: okay fixed it

File: frontend/src/pages/operation_list/individual_list/index.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2020, Verizon Media
```
Comment (JoelAtDeluxe) [inline frontend/src/pages/operation_list/individual_list/index.tsx:1]:
```suggestion
// Copyright 2022, Yahoo Inc.
```

**Review by JoelAtDeluxe — APPROVED** [review]
Looks good to me.

**Comment by jkennedyvz** [comment]
Is it possible to change the page view as you toggle the favorite? (before a reload of /operations)

**Review by jkennedyvz — APPROVED** [review]
Looks great!

**Comment by TylerNoblett** [comment]
alright - I think it's finally done! @jkennedyvz my last two commits don't change the UI; I just cleaned a few things up

**Comment by TylerNoblett** [comment]
alright, I think we're actually good this time

**Review by jkennedyvz — APPROVED** [review]

**Pull request merged**